include/maliput/api/maliput_design.h

```diff
@@ -160,8 +160,60 @@
 ///
 /// @subsubsection scale_length_and_tolerances Scale Length and Tolerances
 ///
-/// > TODO: Explain the concepts of linear tolerance, angular tolerance, and
-/// > characteristic scale length.
+/// The set of abstractions conformed by the `RoadGeometry`, `Junction`, `Segment`,
+/// `Lane` and `BranchPoint` concentrate various aspects of map data, among which we can
+/// identify and highlight:
+/// - Network topology.
+/// - Volume and surface information.
+/// - Dimensions.
+///
+/// Depending on the data source type that will feed an actual implementation of `maliput`
+/// these three aspects will be given, derived or found at the moment of parsing the information
+/// and building the entities. In particular, there is one important characteristic of the
+/// `RoadGeometry` abstraction: the @f$ G^1 @f$ continuity of the road surface. That implies:
+/// - The road surface is continuous.
+/// - The road surface is orientable.
+///
+/// Provided the available map information is imperfect, the numeric system under use
+/// is also imperfect and discrete and there are non-continuities that are by far neglactable
+/// the use of tolerances becomes relevant for designers. The `maliput` API offers
+/// the following tolerance definitions and uses across the codebase:
+///
+/// - *Linear tolerance*: it is a one dimensional tolerance that is used to guarantee @f$ G^0 @f$
+/// continuity in the surfaces and also helps to constrain the expected minimum resolution of
+/// mappings done by implementations.
+/// - *Angular tolerance*: it is a one dimensional tolerance that is used to guarantee @f$ G^1 @f$
+/// continuity in the surfaces, i.e. surfaces are orientable at any point.
+///
+/// Together, these two tolerances constrain the result of geometric queries done via, mostly,
+/// `Lanes`. Their implementations are required to map road volumes that are end to end @f$ G^1 @f$
+/// and connect to other sibling `Lanes` respecting the @f$ G^1 @f$. _Tolerable_ variations in
+/// continuity and orientations are expressed by linear and angular tolerances.
+///
+/// These two tolerances can be further understood as _runtime_ tolerances. Input data, map
+/// data or other procedural generation tools used to build `maliput` backends may contain
+/// noise or they may lack of sufficient data. This leads to another concept, out of the scope of the
+/// `maliput` API concept definition (or distinction), which is the _build time_ tolerance.
```

Comment:
not sure if the concept of _build time_ tolerance add more noise here.

Reply:
In this section I try to define tolerances for what they are and what they are not. I think this approach is useful in the context of this dicussion because everyone has a previous concept of what a tolerance is and what it is used for. The API definition should constrain the applicability and scope of this concept (done with this PR) but also assist users to better understand it. That's why I leave the distinction here.

PTAL in https://github.com/maliput/maliput/commit/0318583c4cff283bf4de0a3f398539d47a43cc69